### ISSUE TITLE
Colon in Node names and edges

### ISSUE THREAD

**Issue by MaxHorwood** [pydot/pydot#224]
Hello,

There are a few other open tickets from a bit ago about this.
Anyway, I was stumbled over it for a bit as I have nodes which are essentially objects which translate with a colon in it. However, this was difficult to debug as the output was still valid as the Node names stripped everything after the colon although the same does not happen for the edges, ie:
```py
# In:
G.add_edge("(thing :test)", "(thing)", label="A Link")
import sys
from networkx.drawing.nx_pydot import write_dot
write_dot(G, sys.stdout)

# Out:
strict graph  {
    "(thing ";
    "(thing)";
    "(thing :test" -- "(thing)"  [label="test1-test2"];
}
```
So I'm using networkx but I know it calls into pydot.to_pydot(...)
The problem is that the edge is different to the node meaning you end up with weird duplicates/nodes that are not attached to anything.

**Comment by ankostis** [comment]
The problem is caused by the "compass point" parsing (see [DOT language](https://www.graphviz.org/doc/info/lang.html)):
https://github.com/pydot/pydot/blob/48ba231b36012c5e13611807c9aac7d8ae8c15c4/pydot.py#L604-L610

Note that the parsing happens even if node-ID is HTML-escaped, so there is definetely a bug here.

To my rough understanding, these should be left as is:

    <foo:bar>

... while this should be treated as `ID:port:compass-point`:

    <foo>:bar:ne
    <foo>:<bar>:sw

**Comment by remi-sap** [comment]
Hi, 
same problem for cluster names, except compilation fails:

```python
import pydot
g = pydot.Dot()
cl1 = pydot.Cluster('14:30')
cl1.add_node(pydot.Node('cafe'))
cl1.add_node(pydot.Node('sieste'))
g.add_subgraph(cl1)
cl2 = pydot.Cluster('17:00')
cl2.add_node(pydot.Node('tea time'))
g.add_subgraph(cl2)
print(g.to_string())

digraph G {
subgraph cluster_14:30 {
cafe;
sieste;
}

subgraph cluster_17:00 {
"tea time";
}

}
```

And this fails when trying to render the graph:
```
g.write_png('cluster with colon.png')

"dot" with args ['-Tpng', '/var/folders/sj/0psyhdm956jcj6xr7nqsp9qh0000gn/T/tmpqg89ae5g'] returned code: 1

stdout, stderr:
 b''
b"Error: /var/folders/sj/0psyhdm956jcj6xr7nqsp9qh0000gn/T/tmpqg89ae5g: syntax error in line 2 near ':'\n"
```
If I manually quote the cluster names then it renders fine. 
` subgraph "cluster_17:00" `

**Comment by jameshfisher** [comment]
I'm guessing this is the same issue:

```
>>> import pydot
>>> dot = pydot.Dot()
>>> dot.add_node(pydot.Node("x:0"))
>>> dot.get_node("x:0")
[]
```

Pydot's job is to hide the dot syntax behind a Python API. As `:` is part of the dot syntax, if a colon is part of a name, pydot should either escape it, or raise an error. Pydot should provide some pure Python way to specify these "ports" and "compass points", e.g. `pydot.Node(("id", "port", "compass_point"))`.

**Comment by peternowee** [comment]
Agreed. pydot/pydot#118 is probably a duplicate of this issue, where the conclusion was the same. I am currently busy looking for work, but hope to pick up pydot work in the evenings once everything stabilizes again. Feel free to open a PR in the mean time.

**Comment by peternowee** [comment]
Btw, I think a colon `:` can be part of a name (ID) in dot actually, but then that ID just needs to be wrapped in double quotes.

**Comment by MaxHorwood** [comment]
I was having a little go at creating a PR. Fixing the issue above is straight forward. However, adding ports and compass points is a bit more involved.
Taking a quick look at how pydot works, it seems like it might be a little finicky to add this functionallity or may cause weird _edge_ cases, such as what happens if you update the compass point after already adding an Edge, how do you then get the compass point when it comes time to making the Edge -> `to_string()` as the Edge only seems to know about the Node ID / doesn't seem to be a way to get the nodes from an Edge etc. I'll try and give this a more concerted effort in a few days if no one beats me to it.
